### PR TITLE
【Prim】Support amp logic for layer_norm and softmax

### PR DESCRIPTION
**Pull request by JiabinYang**
### PR types

Others
### PR changes

Others
### Describe

Pcard-66975
This Pr support amp mode for softmax and layernorm

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_softmax.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def compare_forward(self):
             atol=attrs.get_atol("forward"),
         )
 
-    def test_forward(self):
+    def _test_forward(self):
```
Comment (cyber-pioneer) [inline python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_softmax.py]: debug typo?

File: python/paddle/incubate/autograd/composite_rules.py [hunk 2]
```diff
@@ -34,15 +34,25 @@ def _composite(op, *args):
 @REGISTER_COMPOSITE('softmax')
 def softmax_composite(x, axis):
     """define composite rule of op softmax"""
+    is_amp = False
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py:37]: better to record why not float16

File: python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_softmax.py (hunk 1 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/prim/composite_ops/test_composite_softmax.py]: done

File: python/paddle/incubate/autograd/composite_rules.py (hunk 2 above)
Comment (JiabinYang) [reply python/paddle/incubate/autograd/composite_rules.py:37]: done

**Review by cyber-pioneer — APPROVED** [review]

**Review by xiaoguoguo626807 — APPROVED** [review]
LGTM

**Pull request merged**